rust/src/coordinator.rs

```diff
@@ -19,6 +19,41 @@ use super::{
     PetError,
 };
 
+/// A public key generated by a sum participant. It is used by the
+/// update participants to encrypt their masking seed for each sum
+/// participant.
+type SumParticipantEphemeralPublicKey = box_::PublicKey;
+
+/// A public key that identifies a sum participant.
+type SumParticipantPublicKey = box_::PublicKey;
+
+/// Dictionary created during the first phase of the protocol. It maps
+/// the public key of every sum participant to the ephemeral public
+/// key that participant generated.
+type SumDict = HashMap<SumParticipantPublicKey, SumParticipantEphemeralPublicKey>;
+
+/// Public key that identifies an update participant
+type UpdateParticipantPublicKey = box_::PublicKey;
+
+/// Seed dictionaries are sent by update participants. They contain
+/// the participant's seed, encrypted with the public key of each sum
+/// participant.
+type SeedDict = HashMap<SumParticipantPublicKey, EncryptedMaskingSeed>;
+
+/// The update dictionary is built from the seed dictionaries sent by
+/// the update participants. It maps the encrypted masking seeds of
+/// all the update participants to each sum participant.
+type UpdateDict =
+    HashMap<SumParticipantPublicKey, HashMap<UpdateParticipantPublicKey, EncryptedMaskingSeed>>;
+
+/// Masking seed generated by an update participant, encrypted with
+/// the ephemeral public key of a sum participant
+pub type EncryptedMaskingSeed = Vec<u8>;
+
+/// A 32 bytes hash of that identifies a model mask computed by an
+/// update participant.
```

Comment:
the masks are computed by the sum participants

Reply:
Thank you. I fixed it.